### PR TITLE
Add load more and modify Home Tab

### PR DESCRIPTION
**Pull request by nguyenvanducc**
https://edu-redmine.framgia.vn/issues/1574

### CODE REVIEW

**Comment by nguyenvanducc** [comment]
ready

File: app/src/main/java/com/framgia/nguyenvanducc/soundcloud33/screen/main/genredetail/GenreDetailFragment.java [hunk 1]
```diff
@@ -49,26 +57,53 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+    }
+
+    @Override
+    public void updateFavorite(int position) {
+        mGenreDetailAdapter.updateFavorite(position);
+    }
+
+    @Override
+    public void downloadTrack(boolean isDownloadable, Track track) {
+        if (!isDownloadable) Toast.makeText(getContext()
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/nguyenvanducc/soundcloud33/screen/main/genredetail/GenreDetailFragment.java]: add { } tag

File: app/src/main/java/com/framgia/nguyenvanducc/soundcloud33/screen/main/genredetail/GenreDetailPresenter.java [hunk 2]
```diff
@@ -34,12 +37,32 @@ public void onStop() {
...
+        Track track = mTracks.get(position);
+        if (track.isFavorite()) mTrackRepository.removeFavoriteTrack(track);
+        else mTrackRepository.insertFavoriteTrack(track);
+        track.setFavorite(!track.isFavorite());
+        mView.updateFavorite(position);
+    }
+
+    @Override
+    public void downloadTrack(int position) {
+        boolean isDownloadable =mTracks.get(position).isDownloadable();
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/nguyenvanducc/soundcloud33/screen/main/genredetail/GenreDetailPresenter.java]: format code

File: app/src/main/java/com/framgia/nguyenvanducc/soundcloud33/utils/EndlessRecyclerViewScrollListener.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    // before loading more.
+    private int visibleThreshold = 5;
+    // The current offset index of data you have loaded
+    private int currentPage = 0;
+    // The total number of items in the dataset after the last load
+    private int previousTotalItemCount = 0;
+    // True if we are still waiting for the last set of data to load.
+    private boolean loading = true;
+    // Sets the starting page index
+    private int startingPageIndex = 0;
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/nguyenvanducc/soundcloud33/utils/EndlessRecyclerViewScrollListener.java]: 10 đến 19, copy code quên convention này (tat5)

**Pull request merged**